addons/api/addon/serializers/host-catalog.js

```diff
@@ -68,15 +68,36 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { compositeType, credentialType, isPlugin } = snapshot.record;
     const { options } = attribute;
-
     const fields =
       compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE
         ? azureFields
         : AWSfieldsWithCredentialType[credentialType];
 
-    // Delete any nested attribute fields that don't belong to the record type
+    // Delete any nested attribute fields that don't belong to the record's compositeType
     if (isPlugin && options.isNestedAttribute && json.attributes) {
-      if (!fields.includes(key)) {
+      // If compositeType is AWS, delete azure fields from the json
+      if (compositeType === TYPE_HOST_CATALOG_PLUGIN_AWS) {
+        if (
```

Comment:
I see we are checking twice the condition `!AWSfieldsWithCredentialType[credentialType].includes(key)` alongside with a second condition.

Could this be simplified to check such condition only once?
i.e:

```javascript
if (compositeType === TYPE_HOST_CATALOG_PLUGIN_AWS) {
  if (!AWSfieldsWithCredentialType[credentialType].includes(key)) {
    if (azureFields.includes(key)) {
      delete json.attributes[key];
    }
    json.attributes[key] = null;
  }
}
```
What do you think?

Reply:
💯 this works here, I just added `else` condition to this block. updated!